### PR TITLE
Tech-debt: Rubocop tidy

### PR DESCRIPTION
**Pull request by colinbruce**
## What

Semi-regular attempt to reduce rubocop todo's and disables

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by colinbruce** [comment]
As I'll be on leave if someone does review this, can they merge it too?
Ta :+1:

File: app/models/ccms/submission.rb [hunk 1]
```diff
@@ -16,24 +16,24 @@ class Submission < ApplicationRecord
...
+      'case_ref_obtained' => CCMS::Submitters::ObtainApplicantReferenceService,
+      'applicant_submitted' => CCMS::Submitters::CheckApplicantStatusService,
+      'applicant_ref_obtained' => CCMS::Submitters::ObtainDocumentIdService,
+      'document_ids_obtained' => CCMS::Submitters::AddCaseService,
+      'case_submitted' => CCMS::Submitters::CheckCaseStatusService,
+      'case_created' => CCMS::Submitters::UploadDocumentsService
+    }.freeze
+
+    def process!(options = {})
+      service = STATE_SERVICES[aasm_state.to_s]
```
Comment (Alex-Swann) [inline app/models/ccms/submission.rb:30]: We could do `aasm_state.to_sym` if we don't know if `aasm_state` comes in as a sym which might be nicer as then we could convert all `STATE_SERVICES` keys to syms instead

File: app/services/ccms/payload_generators/entity_attributes_generator.rb [hunk 2]
```diff
@@ -56,18 +56,36 @@ def boolean?(value)
...
-        when 'text', 'number', 'boolean'
-          raw_value
-        when 'currency'
-          as_currency(raw_value)
-        when 'date'
-          raw_value.is_a?(Date) ? raw_value.strftime('%d-%m-%Y') : raw_value
-        else
-          raise CCMSError, "Submission #{submission.id} - Unknown response type in attributes config yaml file: #{config[:response_type]}"
-        end
+        method_to_call = RESPONSE_TYPES[config[:response_type]]
```
Comment (Alex-Swann) [inline app/services/ccms/payload_generators/entity_attributes_generator.rb:85]: sym response_type and use RESPONSE_TYPES keys as syms

File: app/services/ccms/payload_generators/entity_attributes_generator.rb [hunk 3]
```diff
@@ -56,18 +56,36 @@ def boolean?(value)
         value.is_a?(TrueClass) || value.is_a?(FalseClass)
       end
 
+      # def extract_response_value(config)
```
Comment (Alex-Swann) [inline app/services/ccms/payload_generators/entity_attributes_generator.rb:59]: I'm guessing committed in error and should be removed?

**Review by Alex-Swann — CHANGES_REQUESTED** [review]
A few minor comments to address.

**Comment by mpw5** [comment]
> A few minor comments to address. Might be worth someone with more in-depth rubucop knowledge reviewing and confirming they are happy with this also. If so, I can fix my own comments and get this out the door. @stephenrichards, @mpw5, @willc-work ?

@Alex-Swann This all looks good to me. Feel free to make the changes you suggested and I'll re-review when you're done.

**Review by Alex-Swann — APPROVED** [review]

**Review by mpw5 — APPROVED** [review]
LGTM

**Pull request merged**